src/ripple/basics/Slice.h

```diff
@@ -117,6 +126,21 @@ class Slice
     }
     /** @} */
 
+    /** Shrinks the slice by moving its start forward by n characters. */
+    void
+    remove_prefix(std::size_t n)
+    {
+        data_ += n;
+        size_ -= n;
+    }
+
+    /** Shrinks the slice by moving its end backward by n characters. */
+    void
+    remove_suffix(std::size_t n)
+    {
```

Comment:
Consider making it constexpr void remove_suffix(std::size_t n) with noexcept if you decide not to throw.